File: library/src/test/java/io/ashdavies/rx/rxtasks/CompletableTaskListenerFactoryTest.java
package io.ashdavies.rx.rxtasks;

import io.reactivex.CompletableEmitter;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import static com.google.common.truth.Truth.assertThat;

@RunWith(MockitoJUnitRunner.class)
public class CompletableTaskListenerFactoryTest {

  private TaskListenerFactory<Void, CompletableEmitter> factory;

  @Mock CompletableEmitter emitter;

  @Before
  public void setUp() throws Exception {
    factory = new CompletableTaskListenerFactory();
  }

  @Test
  public void shouldReturnCompletableEmitterSuccessListener() throws Exception {
    assertThat(factory.createOnSuccessListener(emitter)).isInstanceOf(CompletableEmitterSuccessListener.class);
  }

  @Test
  public void shouldReturnCompletableEmitterFailureListener() throws Exception {
    assertThat(factory.createOnFailureListener(emitter)).isInstanceOf(CompletableEmitterFailureListener.class);
  }
}

File: library/src/test/kotlin/io/ashdavies/rx/rxtasks/CompletableTaskListenerFactoryTest.kt
package io.ashdavies.rx.rxtasks

import com.google.common.truth.Truth.assertThat
import com.nhaarman.mockito_kotlin.mock
import io.reactivex.CompletableEmitter
import org.junit.Test

internal class CompletableTaskListenerFactoryTest {

  private val emitter = mock<CompletableEmitter>()
  private val factory = CompletableTaskListenerFactory()

  @Test
  fun `should return completable emitter success listener`() {
    assertThat(factory.createOnSuccessListener(emitter)).isInstanceOf(CompletableEmitterSuccessListener::class.java)
  }

  @Test
  fun `should return completable emitter failure listener`() {
    assertThat(factory.createOnFailureListener(emitter)).isInstanceOf(CompletableEmitterFailureListener::class.java)
  }
}
